server/app.mjs

```diff
@@ -361,6 +317,25 @@ const mount = async (
     socket.emit('toolRemoved', state.tools);
   });
 
+  socket.on('saveScript', async () => {
+    if (runningScript) {
+      await runningScript.close();
+      runningScript = null;
+    }
+
+    state.tools = state.tools.filter((t) => t === knowledgeTool);
```

Comment:
Nvm, looks like you are filtering out the knowledge tool. I wonder if you can just get the value from code instead of passing around through the socket.

Reply:
It wasn't letting me import the package from app.mjs, so this was the best I could do.